src/xia2/Wrappers/XDS/XDS.py

```diff
@@ -253,18 +253,17 @@ def imageset_to_xds(
     # based on wavelength. May be useful to override this for in vacuo exps
     # result.append('AIR=0.001')
 
-    if detector == "PILATUS":
+    if detector in ("PILATUS", "EIGER"):
         try:
             thickness = converter.get_detector()[0].get_thickness()
-            if not thickness:
-                thickness = 0.32
-                logger.debug(
-                    "Could not determine sensor thickness. Assuming default PILATUS 0.32mm"
-                )
         except Exception:
-            thickness = 0.32
+            thickness = None
+        if not thickness:
+            thickness = {"PILATUS": 0.32, "EIGER": 0.45}.get(detector)
```

Comment:
Without this line here, does the test above use the correct sensor thickness? i.e. we _do_ get the right value from the header and setting the default here is just for convenience.

Reply:
The correct value is present:

```
Grey-Area cctbx :) $ h5ls -rvd /Users/graeme/svn/cctbx/build/dials_data/vmxi_thaumatin/image_15799_master.h5/entry/instrument/detector/sensor_thickness
Opened "/Users/graeme/svn/cctbx/build/dials_data/vmxi_thaumatin/image_15799_master.h5" with sec2 driver.
entry/instrument/detector/sensor_thickness Dataset {SCALAR}
    Attribute: units scalar
        Type:      3-byte null-terminated ASCII string
        Data:  "mm"
    Location:  1:15832
    Links:     1
    Storage:   8 logical bytes, 8 allocated bytes, 100.00% utilization
    Type:      native double
    Data:
        (0) 0.45
```

Reply:
Additional: CdTe has sensor thickness of 750µm not 450µm

https://www.dectris.com/products/eiger2/eiger2-x-xe-cdte-for-synchrotron/eiger2-x-cdte-4m/

Reply:
The try/except fallback to defaults was there already - I just updated it to differentiate between PILATUS and EIGER. I'm not sure under what circumstances we would get an exception here, but presumably it is there for a reason...

Reply:
Not unheard of for this to be absent from headers because reasons -> error -> default is better estimate than 0...

I did wonder if we should actually _remove_ this block and instead fail...